metabase/metabase
# add pie.decimal_places setting

Closes https://github.com/metabase/metabase/issues/12152

### Description

Adds a viz setting for the pie chart to allow setting a fixed number of decimals places when displaying percentages either on the chart or legend. If the setting is not present, we will use the old logic, which is to use 2 significant digits (e.g. number of digits after the leftmost zero) on the chart, and 3 significant digits on the legend.

### How to verify

1. Create a pie chart
2. Enable the show percentages on legend setting
3. Set the value of the number of decimal places setting
4. Confirm the legend matches your setting
5. Change the setting to show percentages on teh chart
6. Confirm the chart matches your setting

### Demo

https://github.com/user-attachments/assets/38968503-16d8-4bdb-95ba-a549ed55a639

### Checklist
